src/Reflection/ReflectionClass.php

```diff
@@ -775,8 +775,12 @@ public function getParentClass() : ?ReflectionClass
         }
 
         // @TODO use actual `ClassReflector` or `FunctionReflector`?
-        /** @var self $parent */
-        $parent = $this->reflector->reflect($this->node->extends->toString());
+        if ($this->isAnonymous()) {
+            $parent = self::createFromName($this->node->extends->toString());
```

Comment:
This seems incorrect: what happens if you try to get the `ReflectionProperty` for `$someProperty` (defined in `ClassForHinting`) in your test? If I get it correctly, it won't be defined with this code.

Reply:
To be more precise, `assertEquals($parent, $reflector->reflector(ParentName::class))`

Reply:
I improved the test to check for the property and it seems to work, I also added a `\PhpParser\NodeVisitor\NameResolver` to the `AnonymousClassObjectSourceLocator` to resolve the parent class name.

Reply:
Checked this again: it won't use the same reflector configuration when calling `self::createFromName()`. 

The tests work because the defaults of `(new BetterReflection())` use the `ComposerAutoloader` (IIRC), but I was indeed right to smell something here :D

Ideally, `$this->reflector` should always to happen, and `self::createFromName()` should never be used.

I think the changes in `AnonymousClassObjectSourceLocator` are sufficient to make this patch work, or at least they are heading in the right direction: changes in `ReflectionClass` (here) shouldn't be needed.

Reply:
If I'm understanding what's going on correctly, it *can't* use the same reflector, since the reflector used when doing `\Roave\BetterReflection\Reflection\ReflectionObject::createFromInstance` will ONLY use the located source for `\Roave\BetterReflection\SourceLocator\Type\AnonymousClassObjectSourceLocator`:

https://github.com/Roave/BetterReflection/blob/311c6c98de57807ac186f55d0c3dfa96971fa8f0/src/Reflection/ReflectionObject.php#L66-L70

I think this means the parent class isn't in scope, so any class located using that instance of the reflector is going to be the anonymous one (hence why, if you remove the change here, it will return the wrong value:

```
Failed asserting that two strings are identical.
Expected :'Roave\BetterReflectionTest\Fixture\ClassForHinting'
Actual   :'class@anonymous /home/james/workspace/better-reflection/test/unit/Reflection/ReflectionClassTest.php(694)'
```

Possible ideas for fixing, let's discuss before implementing though (I think the right place to fix this is in `\Roave\BetterReflection\Reflection\ReflectionObject::createFromInstance`?)

 * Instead of ONLY using `\Roave\BetterReflection\SourceLocator\Type\AnonymousClassObjectSourceLocator`, wrap it up with an aggregate source locator of `\Roave\BetterReflection\BetterReflection::sourceLocator()` combined with `AnonymousClassObjectSourceLocator`
 * Add a second parameter to `createFromInstance` to inject an existing source locator
 * Is there some other way of getting an existing (or rather "the" existing source locator) in...?